orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -0,0 +1,559 @@
+		}
+		try {
+			boolean existsPrimaryKey = EntityKeyUtil.existsPrimaryKeyInDB( provider.getConnection(), key );
+			log.info( "insertOrUpdateTuple:Key:" + dbKeyName + " exists in database ? " + existsPrimaryKey );
+
+			if ( existsPrimaryKey ) {
+				// it is update
+				queryBuffer.append( "update  " ).append( key.getTable() ).append( "  set " );
+
+				for ( String columnName : tuple.getColumnNames() ) {
+					if ( OrientDBConstant.SYSTEM_FIELDS.contains( columnName ) || columnName.equals( dbKeyName ) ) {
+						continue;
+					}
+					// @TODO correct type
+					queryBuffer.append( " " ).append( columnName ).append( "=" );
+					EntityKeyUtil.setFieldValue( queryBuffer, tuple.get( columnName ) );
+					queryBuffer.append( "," );
+				}
+				queryBuffer.setLength( queryBuffer.length() - 1 );
+				queryBuffer.append( " WHERE " ).append( dbKeyName ).append( "=" );
+				EntityKeyUtil.setFieldValue( queryBuffer, dbKeyValue );
+			}
+			else {
+				// it is insert with business key which set already
+				log.info( "insertOrUpdateTuple:Key:" + dbKeyName + " is new! Insert new record!" );
+				queryBuffer.append( "insert into " ).append( key.getTable() ).append( "  set " );
+				for ( String columnName : tuple.getColumnNames() ) {
+					if ( OrientDBConstant.SYSTEM_FIELDS.contains( columnName ) ) {
+						continue;
+					}
+					// @TODO correct type
+					queryBuffer.append( " " ).append( columnName ).append( "=" );
+					EntityKeyUtil.setFieldValue( queryBuffer, tuple.get( columnName ) );
+					queryBuffer.append( "," );
+				}
+				queryBuffer.append( dbKeyName ).append( " = " );
+				EntityKeyUtil.setFieldValue( queryBuffer, dbKeyValue );
+			}
+
+			log.info( "insertOrUpdateTuple:Key:" + dbKeyName + " (" + dbKeyValue + ").  query:" + queryBuffer.toString() );
+			PreparedStatement pstmt = connection.prepareStatement( queryBuffer.toString() );
+			log.info( "insertOrUpdateTuple:Key:" + dbKeyName + " (" + dbKeyValue + "). inserted or updated: " + pstmt.executeUpdate() );
+		}
+		catch (SQLException e) {
+			log.error( "Can not find entity", e );
+			throw new RuntimeException( e );
+		}
+	}
+
+	@Override
+	public void insertTuple(EntityKeyMetadata entityKeyMetadata, Tuple tuple, TupleContext tupleContext) {
+		log.info( "insertTuple:EntityKeyMetadata:" + entityKeyMetadata + "; tupleContext" + tupleContext + "; tuple:" + tuple );
+
+		StringBuilder insertQuery = new StringBuilder( 100 );
+		insertQuery.append( "insert into " ).append( entityKeyMetadata.getTable() ).append( " " );
+		if ( !tuple.getColumnNames().isEmpty() ) {
+			insertQuery.append( " set " );
+		}
+
+		String dbKeyName = entityKeyMetadata.getColumnNames()[0];
+		Long dbKeyValue = null;
+		Connection connection = provider.getConnection();
+
+		if ( dbKeyName.equals( OrientDBConstant.SYSTEM_RID ) ) {
+			// use @RID for key
+			throw new UnsupportedOperationException( "Can not use @RID as primary key!" );
```

Comment:
This should be an `HibernateException`, check `org.hibernate.datastore.ogm.orientdb.logging.impl.Log` and create a new error log for it.